Apps/Database/src/Delegates/DBUserDelegateDelegate.cs

```diff
@@ -91,14 +91,9 @@ public DBResult<IEnumerable<UserDelegate>> Get(string delegateId, int page, int
         public DBResult<UserDelegate> Delete(string ownerId, string delegateId, bool commit)
         {
             this.logger.LogTrace($"Deleting UserDelegate (ownerId: {ownerId}, delegateId: {delegateId}) from DB...");
-            UserDelegate userDelegate = new UserDelegate()
```

Comment:
We should give this back to whomever did delete.  We should not be doing a query to do a delete, instead we should be passing in the object that was originally fetched or marginally the key identifiers along with the version number.

Reply:
at this point the whole fetch does not have a version in it, so....

Reply:
UserDelegate is defined as public class UserDelegate : AuditableEntity anything that is Auditable has a version attached to it.  The object returned by GetUserDelegate and InsertUserDelegate is returning the full object.  We should be using that...

Reply:
I've updated the model to contain a representation of the user delegate to correctly use the concurrency checks